gm4_blossoming_pots/data/gm4_blossoming_pots/function/flower/as_marker.mcfunction

```diff
@@ -0,0 +1,46 @@
+# Sets up loop for plant stage changes
+# @s = temp marker inside flower pot
+# at @s align xyz positioned ~.5 ~.5 ~.5
+# with {UUID, id, rotation}
+# run from flower/as_player
+
+# early returns
+# if empty hand, empty the pot, give player the item
+execute \
+    if data storage gm4_blossoming_pots:flower_pots {temp:{id:"null"}} \
+    run return run function gm4_blossoming_pots:flower/item/give_back with storage gm4_blossoming_pots:flower_pots temp
+# if storage doesn't have, doesnt matter
+$execute unless data storage gm4_blossoming_pots:flower_pots $(id) run return run kill @s
+# if flower pot is full, return
+execute if data entity @n[type=minecraft:marker,tag=gm4_blossoming_pots.data.flower_pot,distance=..0.2] {data:{full:1}} run return run kill @s
+
+# if no perma marker yet, summon it
+$execute unless entity @n[type=minecraft:marker,tag=gm4_blossoming_pots.data.flower_pot,distance=..0.2] \
+    run summon minecraft:marker ~ ~ ~ {Tags:["gm4_blossoming_pots.data.flower_pot","smithed.strict","smithed.entity"],data:{id:"$(id)",count:0,full:0},CustomName:'{"text":"gm4_blossoming_pots.flower_pot"}'}
+
+# decrement player mainhand by 1
+$execute as @p[nbt={UUID:$(UUID)},gamemode=!creative] run item modify entity @s weapon.mainhand gm4_blossoming_pots:count_decr
+# increment perma marker count by 1
+    # this is a nightmare of selectors
+        # could tag it at the start of the function to have this be just checking that one tag?
+execute store result score @n[type=minecraft:marker,tag=gm4_blossoming_pots.data.flower_pot,distance=..0.2] gm4_blossoming_pots.loop \
+    run data get entity @n[type=minecraft:marker,tag=gm4_blossoming_pots.data.flower_pot,distance=..0.2] data.count 1
+scoreboard players add @n[type=minecraft:marker,tag=gm4_blossoming_pots.data.flower_pot,distance=..0.2] gm4_blossoming_pots.loop 1
+execute store result entity @n[type=minecraft:marker,tag=gm4_blossoming_pots.data.flower_pot,distance=..0.2] data.count int 1 \
+    run scoreboard players get @n[type=minecraft:marker,tag=gm4_blossoming_pots.data.flower_pot,distance=..0.2] gm4_blossoming_pots.loop
```

Comment:
These should be moved into a function executed as `@n[type=minecraft:marker,tag=gm4_blossoming_pots.data.flower_pot,distance=..0.2]`, so you can replace the selectors with `@s`